src/main/java/org/icatproject/core/manager/search/SearchManager.java

```diff
@@ -145,25 +206,68 @@ private class PendingSearchRequestHandler extends TimerTask {
 
 		@Override
 		public void run() {
-			synchronized (backlogHandlerFileLock) {
-				if (backlogHandlerFile.length() != 0) {
-					logger.debug("Will attempt to process {}", backlogHandlerFile);
-					try (BufferedReader reader = new BufferedReader(new FileReader(backlogHandlerFile))) {
-						String line;
-						while ((line = reader.readLine()) != null) {
-							searchApi.modify(line);
+			int numLines = 0;
+
+			while (numLines < backlogLinesPerTimer) {
+				synchronized (backlog.getReadLock()) {
+					Path path;
+					try {
+						path = backlog.getReadPath();
+					} catch (IcatException e) {
+						// Already logged
+						return;
+					}
+
+					if (path == null) {
+						logger.trace("No backlog file available to process");
+						break;
+					}
+
+					Path dotnewPath = Paths.get(path + ".new");
+
+					logger.debug("Will attempt to process {}", path);
+
+					String line;
+					try (BufferedReader reader = Files.newBufferedReader(path)) {
+						line = reader.readLine();
+						if (line == null) {
+							break;
+						}
+
+						try (BufferedWriter writer = Files.newBufferedWriter(dotnewPath, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+							String line2;
+							while ((line2 = reader.readLine()) != null) {
+								writer.write(line2);
+								writer.newLine();
+							}
+						} catch (IOException e) {
+							logger.error("Error writing to {}", dotnewPath, e);
+							return;
 						}
```

Comment:
Same as the point above, is the impact of doing this in the while loop small compared to the benefit of keeping the file on disk up to date/whatever other reasons we have for doing it this way?